src/verilog.y

```diff
@@ -4901,22 +5000,66 @@ gatePinExpr<nodep>:
-        ;
+strength0<nodep>:
+                ySUPPLY0                                { $$ = new AstStrength($1, AstStrength::SUPPLY, 0); }
+| ySTRONG0                                { $$ = new AstStrength($1, AstStrength::STRONG, 0); }
+| yPULL0                                { $$ = new AstStrength($1, AstStrength::PULL, 0); }
+| yWEAK0                                { $$ = new AstStrength($1, AstStrength::WEAK, 0); }
+;
+
+strength1<nodep>:
+                ySUPPLY1                                { $$ = new AstStrength($1, AstStrength::SUPPLY, 1); }
+| ySTRONG1                                { $$ = new AstStrength($1, AstStrength::STRONG, 1); }
+| yPULL1                                { $$ = new AstStrength($1, AstStrength::PULL, 1); }
+| yWEAK1                                { $$ = new AstStrength($1, AstStrength::WEAK, 1); }
+;
+
+driveStrengthE<nodep>:
+/* empty */                             { $$ = nullptr; }
+|       driveStrength                            { $$ = $1; }
+        ;
+
+
+driveStrength<nodep>:
+yP_PAR__STRENGTH strength0 ',' strength1 ')' { $$ = new AstStrengthSpec($1, VN_AS($2, Strength), VN_AS($4, Strength)); }
+| yP_PAR__STRENGTH strength1 ',' strength0 ')' { $$ = new AstStrengthSpec($1, VN_AS($4, Strength), VN_AS($2, Strength)); }
+| yP_PAR__STRENGTH strength0 ',' yHIGHZ1 ')' { AstStrength* highz1p = new AstStrength($1, AstStrength::HIGHZ, 1);
+    $$ = new AstStrengthSpec($1, VN_AS($2, Strength), highz1p); }
+| yP_PAR__STRENGTH strength1 ',' yHIGHZ0 ')' { AstStrength* highz0p = new AstStrength($1, AstStrength::HIGHZ, 0);
+    $$ = new AstStrengthSpec($1, highz0p, VN_AS($2, Strength)); }
+| yP_PAR__STRENGTH yHIGHZ0 ',' strength1 ')' { AstStrength* highz0p = new AstStrength($1, AstStrength::HIGHZ, 0);
+    $$ = new AstStrengthSpec($1, highz0p, VN_AS($4, Strength)); }
+| yP_PAR__STRENGTH yHIGHZ1 ',' strength0 ')' { AstStrength* highz1p = new AstStrength($1, AstStrength::HIGHZ, 1);
+    $$ = new AstStrengthSpec($1, VN_AS($4, Strength), highz1p); }
+;
+
+pulldownStrengthE<nodep>:
+{ $$ = nullptr; }
+| pulldownStrength { $$ = $1; }
+;
+
+pulldownStrength<nodep>:
+yP_PAR__STRENGTH strength0 ',' strength1 ')' { delete $4;
+    $$ = new AstStrengthSpec($1, VN_AS($2, Strength), nullptr); }
+| yP_PAR__STRENGTH strength1 ',' strength0 ')' { delete $2;
+    $$ = new AstStrengthSpec($1, VN_AS($4, Strength), nullptr); }
+| yP_PAR__STRENGTH strength0 ')' {
+    $$ = new AstStrengthSpec($1, VN_AS($2, Strength), nullptr); }
+;
+
+pullupStrengthE<nodep>:
+{ $$ = nullptr; }
+| pullupStrength { $$ = $1; }
+;
+
+pullupStrength<nodep>:
+yP_PAR__STRENGTH strength0 ',' strength1 ')' { delete $2;
+    $$ = new AstStrengthSpec($1, nullptr, VN_AS($4, Strength)); }
+| yP_PAR__STRENGTH strength1 ',' strength0 ')' { delete $4;
+    $$ = new AstStrengthSpec($1, nullptr, VN_AS($2, Strength)); }
+| yP_PAR__STRENGTH strength1 ')' {
+    $$ = new AstStrengthSpec($1, nullptr, VN_AS($2, Strength)); }
+;
```

Comment:
Formatting, but I guess you're working on it?

Reply:
I knot that formatting needs to be changed, but I don't know which tool to use and what the configuration should be. Maybe we can just ask Wilson about it. I decided to leave it for now.